app/org/sagebionetworks/bridge/dao/HealthDataDao.java

```diff
@@ -1,24 +1,36 @@
 package org.sagebionetworks.bridge.dao;
 
+import javax.annotation.Nonnull;
 import java.util.List;
 
-import org.sagebionetworks.bridge.models.healthdata.HealthDataKey;
 import org.sagebionetworks.bridge.models.healthdata.HealthDataRecord;
+import org.sagebionetworks.bridge.models.healthdata.HealthDataRecordBuilder;
 
+/** DAO for health data records. */
 public interface HealthDataDao {
+    /**
+     * DAO method used by worker apps to creating a health data record and persisting it, generally from unpacking
+     * uploads.
+     *
+     * @param record
+     *         health data record prototype, from which the record should be created from, must be non-null
+     * @return the unique ID of the created record
+     */
+    String createRecord(@Nonnull HealthDataRecord record);
 
-    public List<HealthDataRecord> appendHealthData(HealthDataKey key, List<HealthDataRecord> records);
```

Comment:
Discussed this offlne with Alx. We're still planning on supporting these features in the future. We're just not implementing these yet, since no one needs them yet. However, Health Data v2 is built in a way that we could add these easily.